### PR TITLE
[K8SPG-165] Optional Resource requests/limits for operator containers

### PR DESCRIPTION
**Pull request by djtaylor**
Cleaned up stray commits from previous attempt at a PR. This provides a way to specify optional resource requests/limits for the containers in the operator. In a Kubernetes cluster where resource quotas are mandatory, these will always fail to start unless specified.

See: https://jira.percona.com/projects/K8SPG/issues/K8SPG-165?filter=allopenissues

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade | failed |
| demand-backup | failed |
| init-deploy | failed |
| scheduled-backup | failed |
| scaling | failed |
| smart-update | failed |
| recreate | failed |
| version-service | failed |
| affinity | failed |
| monitoring | failed |
| self-healing | failed |
| operator-self-healing | failed |
| clone-cluster | failed |
| tls-check | failed |
| users | failed |
| ns-mode | failed |

commit: https://github.com/percona/percona-postgresql-operator/pull/193/commits/96e43a8fdaff4a2861102caa102ef443c460126d
image: `perconalab/percona-postgresql-operator:PR-193-96e43a8f-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-193-96e43a8f-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-193-96e43a8f-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-193-96e43a8f-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-193-96e43a8f-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-193-96e43a8f-pgo-deployer`

File: installers/ansible/roles/pgo-operator/templates/deployment.json.j2 [hunk 1]
```diff
@@ -189,6 +213,17 @@
...
+                        "resources": {
+                          "requests": {
+                            "cpu": "{{ scheduler_resources_requests_cpu }}",
+                            "memory": "{{ scheduler_resources_requests_memory }}"
+                          },
+                          "limits": {
+                            "cpu": "{{ scheduler_resources_limits_cpu }}",
+                            "memory": "{{ scheduler_resources_limits_memory }}"
+                          }
+                        },
```
Comment (hors) [inline installers/ansible/roles/pgo-operator/templates/deployment.json.j2:226]:
```suggestion
                        {% if scheduler_resources == "true" %}
                        "resources": {
                          "requests": {
                            "cpu": "{{ scheduler_resources_requests_cpu }}",
                            "memory": "{{ scheduler_resources_requests_memory }}"
                          },
                          "limits": {
                            "cpu": "{{ scheduler_resources_limits_cpu }}",
                            "memory": "{{ scheduler_resources_limits_memory }}"
                          }
                        },
                        {% endif %}
```

File: installers/ansible/roles/pgo-operator/templates/deployment.json.j2 [hunk 2]
```diff
@@ -250,6 +285,17 @@
...
+                        "resources": {
+                          "requests": {
+                            "cpu": "{{ event_resources_requests_cpu }}",
+                            "memory": "{{ event_resources_requests_memory }}"
+                          },
+                          "limits": {
+                            "cpu": "{{ event_resources_limits_cpu }}",
+                            "memory": "{{ event_resources_limits_memory }}"
+                          }
+                        },
```
Comment (hors) [inline installers/ansible/roles/pgo-operator/templates/deployment.json.j2:298]:
```suggestion
                        {% if event_resources == "true" %}
                        "resources": {
                          "requests": {
                            "cpu": "{{ event_resources_requests_cpu }}",
                            "memory": "{{ event_resources_requests_memory }}"
                          },
                          "limits": {
                            "cpu": "{{ event_resources_limits_cpu }}",
                            "memory": "{{ event_resources_limits_memory }}"
                          }
                        },
                        {% endif %}
```

File: installers/helm/values.yaml [hunk 3]
```diff
@@ -141,3 +141,27 @@ storage9_access_mode: "ReadWriteOnce"
 storage9_size: "1Gi"
 storage9_type: "dynamic"
 storage9_class: "rook-ceph-block"
+
```
Comment (hors) [inline installers/helm/values.yaml:144]: @cap1984 should we add it?

File: installers/ansible/values.yaml [hunk 4]
```diff
@@ -121,3 +121,27 @@ storage9_access_mode: "ReadWriteOnce"
 storage9_size: "1Gi"
 storage9_type: "dynamic"
 storage9_class: "rook-ceph-block"
+
```
Comment (hors) [inline installers/ansible/values.yaml:124]: @djtaylor please add these lines into `./deploy/operator.yaml` too

**Review by hors — CHANGES_REQUESTED** [review]

File: installers/helm/values.yaml (hunk 3 above)
Comment (cap1984) [reply installers/helm/values.yaml:144]: @djtaylor for helm please check out https://github.com/percona/percona-helm-charts.
We consider removing crunchy helm from this very repo

**Comment by cap1984** [comment]
@djtaylor please follow @hors suggestions.
Baseline looks good to me

**Pull request closed without merging**